src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,227 @@
+
+/**
+ * Utility used by Duke to manage tasks.
+ *
+ * @see Duke
+ * @see Task
+ * @see Storage
+ */
+public class TaskList {
+  private final List<Task> tasks;
+  private final Storage storage = new Storage();
+
+  /**
+   * Constructs a new TaskList with the given tasks.
+   *
+   * @param tasks the tasks to be managed
+   */
+  public TaskList(List<Task> tasks) {
+    this.tasks = tasks;
+    storage.save(tasks);
+  }
+
+  /**
+   * Constructs a new TaskList with no tasks and stores it in the given file.
+   *
+   * @param fileName the name of the file to be used to store tasks.
+   * @throws IOException            if the file cannot be found or created.
+   * @throws ClassNotFoundException if the file cannot be deserialized.
+   */
+  public TaskList(String fileName) throws IOException, ClassNotFoundException, ClassCastException {
+    List<Task> tasks = new ArrayList<>();
+    try {
+      tasks = storage.load(fileName);
+    } catch (FileNotFoundException e) {
+      storage.save(tasks, fileName);
+    }
+    this.tasks = tasks;
+  }
+
+  /**
+   * Constructs a new TaskList with no tasks and stores it in the default location.
+   *
+   * @throws IOException            if the file cannot be found or created.
+   * @throws ClassNotFoundException if the file cannot be deserialized.
+   * @see Storage
+   */
+  public TaskList() throws IOException, ClassNotFoundException, ClassCastException {
+    List<Task> tasks = new ArrayList<>();
+    try {
+      tasks = storage.load();
+    } catch (FileNotFoundException e) {
+      storage.save(tasks);
+    }
+    this.tasks = tasks;
+  }
+
+  /**
+   * Returns a string representation of the tasks in the list.
+   *
+   * @return a string representation of the tasks in the list.
+   * @see Ui#print(String)
+   */
+  public String[] stringify() {
+    if (this.tasks.size() == 0) {
+      return new String[]{"No tasks found."};
+    }
+    String[] outputs = new String[this.tasks.size()];
+    for (int i = 0; i < this.tasks.size(); i++) {
+      outputs[i] = addOrdinal(i, this.tasks.get(i));
+    }
+    return outputs;
+  }
+
+  /**
+   * Creates a new {@link Task} from the given arguments and adds it to the list.
+   *
+   * @param arguments the arguments to be used to create the task.
+   * @return a string representation of the task that was added.
+   */
+  public String[] add(String[] arguments) {
+    Task newTask;
+    switch (arguments[0]) {
+      case "/todo" -> newTask = new ToDo(arguments[1]);
+      case "/deadline" -> {
+        String[] deadlineArgs = Parser.parseArgs(arguments[1], new String[]{" /by "});
+        newTask = new Deadline(deadlineArgs[0], deadlineArgs[1]);
+      }
+      case "/event" -> {
+        String[] eventArgs = Parser.parseArgs(arguments[1], new String[]{" /from ", " /to "});
+        newTask = new Event(eventArgs[0], eventArgs[1], eventArgs[2]);
+      }
+      default -> throw new IllegalArgumentException("Invalid task type: " + arguments[0]);
+    }
```

Comment:
According to the coding standards, there should be no indentation for the `case` clauses.